runelite-client/src/main/java/net/runelite/client/plugins/hpandprayerstatus/HpAndStatusOverlay.java

```diff
@@ -0,0 +1,193 @@
+ */
+package net.runelite.client.plugins.hpandprayerstatus;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.game.SpriteManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import javax.inject.Inject;
+
+@Slf4j
+public class HpAndStatusOverlay extends Overlay
+{
+	private Client c;
+	private HpAndPrayerConfig config;
+	private SpriteManager sm;
+
+	final private int HEIGHT = 224;
+	final private int WIDTH = 15;
+	final private int PADDING = 1;
+	final private int IMAGE_SIZE = 17;
+	// Static colors
+	final Color DEPLETED = new Color(25, 25, 25);
+	final Color PRAYER_LEFT = new Color(50, 200, 200);
+	final Color BACKGROUND = new Color(0, 0 , 0);
+	// Dynamic colors
+	Color HEALTH_LEFT = new Color(200, 35, 0);
+	Color POISONED_HEALTH_LEFT = new Color(0, 145, 0);
+	Color VENOMED_HEALTH_LEFT = new Color(0, 65, 0);
+	Color COUNTER_COLOR = new Color(0, 0, 0, 255);
+
+	@Inject
+	public HpAndStatusOverlay(Client c, SpriteManager sm, HpAndPrayerConfig config)
+	{
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.c = c;
+		this.sm = sm;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		renderInventoryBars(g);
+		return null;
+	}
+
+	private void renderInventoryBars(Graphics2D g)
+	{
+		//Variables
+		int max_health = c.getRealSkillLevel(Skill.HITPOINTS);
+		int max_prayer = c.getRealSkillLevel(Skill.PRAYER);
+		int current_health = c.getBoostedSkillLevel(Skill.HITPOINTS);
+		int current_prayer = c.getBoostedSkillLevel(Skill.PRAYER);
+		int poison_state = c.getVar(VarPlayer.IS_POISONED);
+		int counter_health = c.getBoostedSkillLevel(Skill.HITPOINTS);
+		int counter_prayer = c.getBoostedSkillLevel(Skill.PRAYER);
+		//Store images
+		BufferedImage healthImage = sm.getSprite(SpriteID.SKILL_HITPOINTS, 0);
+		BufferedImage prayerImage = sm.getSprite(SpriteID.SKILL_PRAYER, 0);
+		BufferedImage chatboxImage = sm.getSprite(SpriteID.CHATBOX, 0);
+
+		//Code for counters
+		if (config.enableCounter())
+		{
+			healthImage = chatboxImage;
+			prayerImage = chatboxImage;
+			COUNTER_COLOR = new Color ( 0, 0, 0,  255);
+		}
+		else
+			{
+			COUNTER_COLOR = new Color ( 0, 0, 0,  0);
+		}
+
+		// If the player gets inflicted with poison or venom the colors will be replaced to indicate this status.
+		if (poison_state > 0 && poison_state < 37)
+		{
+			HEALTH_LEFT = POISONED_HEALTH_LEFT;
+		}
+		else if (poison_state > 1000000)
+		{
+			HEALTH_LEFT = VENOMED_HEALTH_LEFT;
+			POISONED_HEALTH_LEFT = VENOMED_HEALTH_LEFT;
+		}
+		else
+		{
+			//reset colors if not poisoned anymore.
+			HEALTH_LEFT = new Color(200, 35, 0);
+			POISONED_HEALTH_LEFT = new Color(0, 145, 0);
+		}
+
+		for (Viewport viewport : Viewport.values())
```

Comment:
Would be nice if you would add the logic for the viewports when an InterfaceStyle skin is on ;)

Reply:
What would be the benefit if i did this? there are no issues when i switch between 2005 / 2010 / normal skins. So is it really needed to detect whether a skin is applied?

Reply:
Ah if that's the case then ignore my comment. I thought it was still the same situation as before.

Reply:
not anymore, that was because i was directly drawing on specific coords which makes it hard to let it change dynamically, which is not the case anymore now with the rewritten code 👍